modules/ximgproc/src/edgepreserving_filter.cpp

```diff
@@ -0,0 +1,387 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+//                    Created by Simon Reich
+//
+
+
+#include "precomp.hpp"
+
+namespace cv
+{
+namespace ximgproc
+{
+using namespace std;
+
+	edgepreservingFilter(const InputArray *src, OutputArray *dst, int d,
+	                         int threshold)
+	{
+
+		CV_Assert(src->type() == CV_8UC3 && src->data != dst->data);
+
+		if (d < 3)
+			d = 3;
+		int subwindowX = d, subwindowY = d;
+
+		if (threshold < 0)
+			threshold = 0;
+
+		// number of image channels
+		int nChannel = src->channels();
+
+		src->copyTo(*dst);
+
+		vector<double> pixel(nChannel, 0);
+		vector<vector<double>> line1(src->rows, pixel);
+		vector<vector<vector<double>>> weight(src->cols,
```

Comment:
you need an extra space between the `>` brackets  here (and in the line below):
     `vector<vector<double> >`
(else older c++ compilers choke on it)